content/ui-patterns/overlays.mdx

```diff
@@ -0,0 +1,434 @@
+
+Make sure the contents of any overlay fit narrow viewports with a width of at least 320px.
+
+Overlays width and height dimensions should be set by the values in the following tables.
+
+<Grid gridTemplateColumns={['1fr', null, null, null, 'auto 1fr']} gridGap={4}>
+<div>
+
+Width     | Size
+:---------|:------
+`small`   | 256px
+`medium`  | 320px
+`large`   | 480px
+`xlarge`  | 640px
+`xxlarge` | 960px
+
+</div>
+<div>
+
+Height   | Size               
+:--------|:-------------------
+`auto`   | Adjusts to content
+`xsmall` | 192px
+`small`  | 256px
+`medium` | 320px
+`large`  | 432px
+`xlarge` | 600px
+
+</div>
+</Grid>
+
+If the height is set to `auto`, make sure the content region has a `max-height` property defined to support overflow scrolling. For overlays that list user-generated information in a single line, use the `large` height for reference.
+
+Avoid creating overlays that are too short or too tall compared to their width. As a general guideline, keep a maximum size ratio of 1.6:1, which translates to the following ranges:
+
+&ZeroWidthSpace;<span style="white-space: nowrap"><sub>Height</sub> ∖ <sup>Width</sup></span> | `small` | `medium` | `large` | `xlarge` | `xxlarge` |
+----------|:-----:|:------:|:-----:|:-----------:|:-----------------:|
+`xsmall`  | ✅    | ✅     | ❌     | ❌          | ❌                 |
+`small`   | ✅    | ✅     | ❌     | ❌          | ❌                 |
+`medium`  | ✅    | ✅     | ✅     | ❌          | ❌                 |
+`large`   | ❌    | ✅     | ✅     | ✅          | ❌                 |
+`xlarge`  | ❌    | ❌     | ✅     | ✅          | ✅                 |
+
+
+## Behaviors
+
+### Opening
+
+Overlays appear on the page after a direct user interaction, such as on click, tap, press, etc.
+
+<!--
+The following paragraph needs a revision based on
+https://github.com/github/primer/issues/575
+
+On desktops and other devices with a fine pointer, overlays may also be triggered by a delayed `hover` event (GitHub calls these _hovercards_). Since this behavior doesn't work on touch devices, consider whether tapping on the source element should open the overlay or take the user to its destination directly. For example, a delayed hover on a **@username** may present a _hovercard_ on desktop, while a tap on the username would open a bottom sheet with a link to the user's profile page.
+-->
+
+
+
+Don't show overlays on page load. For that, prefer in-page messages and flash banners.
+
+<!--
+### Loading
+Pending.
+-->
+
+### Dismissing
+
+Users should be able to dismiss an overlay by:
+- Pressing Escape
+- Pressing on a **Close** button
+- Clicking or tapping outside
+- Pressing the browser’s back button
+
+Close buttons are useful as an easy escape route, particularly in larger or more complex overlays, or when the overlay is no longer visually connected to the source that triggered it. Always use close buttons overlays with backdrops.
```

Comment:
While being FR someone had a question around close buttons and if headers in ActionList/Select menu should come with a close button or not. While I think this helps already a lot I do think we're still a bit vague especially if the ActionList/Select menu component makes it optional to have a that close button. 

[Slack thread (internal)](https://github.slack.com/archives/CSGAVNZ19/p1642433763062900)